codalab/worker/worker.py

```diff
@@ -257,9 +255,11 @@ def checkin(self):
                     self.read_run_missing(socket_id)
                 return
             if action_type == 'kill':
-                self.kill(uuid)
+                self.runs[uuid] = self.runs[uuid]._replace(
```

Comment:
Perhaps better to leave the functions `kill` and `mark_finalized` to parallel the other `action_type`s, which have their own functions.